components/search/src/selector.rs

```diff
@@ -1,27 +1,65 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-use crate::{error::Error, SearchApiResult};
+
+use crate::{error::Error, RefinedConfig, SearchApiResult, SearchUserEnvironment};
 use error_support::handle_error;
 
-pub struct SearchEngineSelector();
+/// SearchEngineSelector parses the JSON configuration for
+/// search engines and returns the applicable engines depending
+/// on their region + locale.
+#[derive(Default, uniffi::Object)]
+pub struct SearchEngineSelector {}
 
+#[uniffi::export]
 impl SearchEngineSelector {
+    #[uniffi::constructor]
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Sets the search configuration from the given string. This allows for
+    /// reprocessing of the configuration if it has not changed since the
+    /// previous update, e.g. to optimise test running.
+    #[handle_error(Error)]
+    pub fn set_search_config(&self, _configuration: String) -> SearchApiResult<()> {
+        Err(Error::NotImplemented)
+    }
+
+    /// Clears the search configuration from memory if it is known that it is
+    /// not required for a time.
+    pub fn clear_search_config(&self) {}
+
+    /// Filters the search configuration with the user's given environment,
+    /// and returns the set of engines and parameters that should be presented
+    /// to the user.
     #[handle_error(Error)]
-    pub fn filter_engine_configuration(&self) -> SearchApiResult<()> {
+    pub fn filter_engine_configuration(
```

Comment:
I assume this function is the equivalent for [`fetchEngineConfiguration` ](https://searchfox.org/mozilla-central/rev/7e0ae4372c52b8183d1178132dd6493edb576738/toolkit/components/search/SearchEngineSelector.sys.mjs#233) ? 

I'm wondering if we should keep the same function names for consistency?

Reply:
They're not quite equivalent. `fetchEngineConfiguration` gets the configuration from remote settings and filters it.

This function only filters the configuration.

Reply:
Right, that makes sense because mobile is fetching the config from remote settings themselves. After that, it's going to call this.

Reply:
I think we might add a fetch + filter, but at the moment I'm waiting to see what the remote settings APIs look like before we decide that.